### PR TITLE
Refactor the replication policy destination namespace logic

### PR DESCRIPTION
**Pull request by ywk253100**
Support specifying what part of the repository will be replaced by the provided namespace

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: api/v2.0/swagger.yaml [hunk 1]
```diff
@@ -5954,6 +5954,13 @@ definitions:
       dest_namespace:
         type: string
         description: The destination namespace.
+      dest_namespace_replace_count:
+        type: integer
+        format: int8
+        description: |-
+          Specify how many path components will be replaced by the provided destination namespace.
+          When it is set as -1, legacy mode will be applied.
```
Comment (bitsf) [inline api/v2.0/swagger.yaml:5962]: 没有设置也可以是 legacy

**Review by bitsf — APPROVED** [review]

File: src/controller/replication/flow/stage.go [hunk 2]
```diff
@@ -186,13 +192,38 @@ func getResourceName(res *model.Resource) string {
...
+
+	// legacy logic to keep backward compatibility
+	if replaceCount < 0 {
+		_, rest := util.ParseRepository(repository)
+		return fmt.Sprintf("%s/%s", namespace, rest), nil
+	}
+
+	subs := strings.Split(repository, "/")
+	len := len(subs)
+	if len < int(replaceCount) {
```
Comment (steven-zou) [inline src/controller/replication/flow/stage.go:213]: Use `switch`?

```go
switch {
  case replaceCount ==0:
  case int(replaceCount) == len:
  case int(replaceCount) > 0:
  default:
  
}
```

**Review by steven-zou — COMMENTED** [review]
Check my comments?

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**